src/Helpers/FiltersHelper.php

```diff
@@ -0,0 +1,224 @@
+
+        // Process other filters based on request inputs
+        self::processAdditionalFilters($request);
+
+        if ($request->input('sort') == '-score') {
+            $request->merge(['sort' => '-published_on']);
+        }
+    }
+
+    /**
+     * Sets the required fields for the content retrieval.
+     *
+     * @param string $field The field to be set as required.
+     * @return void
+     */
+    public static function setRequiredFields($field)
+    {
+        self::$requiredFields = array_merge(self::$requiredFields, [$field]);
+    }
+
+    private static function processStatuses($request)
+    {
+        if ($request->has('statuses')) {
+            ContentRepository::$availableContentStatues = $request->input('statuses');
+        }
+    }
+
+    private static function processFutureContentOptions($request)
+    {
+        if ($request->has('include_future_content')) {
+            ContentRepository::$pullFutureContent = $request->input('include_future_content');
+        }
+
+        if ($request->input('include_future_scheduled_content_only') !== 'false') {
+            ContentRepository::$pullFutureContent = true;
+            self::$futureScheduledContentOnly = true;
+        }
+
+        if ($request->has('future')) {
+            ContentRepository::$pullFutureContent = true;
+        }
+    }
+
+    private static function processCountFilterItems($request)
+    {
+        if ($request->has('count_filter_items')) {
+            ContentRepository::$countFilterOptionItems = 1;
+        }
+    }
+
+    private static function processEnrollmentOptions($request)
+    {
+        if ($request->has('without_enrollment')) {
+            ContentRepository::$getEnrollmentContent = !$request->input('without_enrollment');
+        }
+
+        if ($request->input('only_from_my_list') == "true") {
+            $userPlaylistsService = app()->make(UserPlaylistsService::class);
+            $myList = $userPlaylistsService->getUserPlaylist(
+                user()->id,
+                'user-playlist',
+                config('railcontent.brand')
+            );
+            ContentRepository::$includedInPlaylistsIds =
+                $myList->pluck('id')
+                    ->toArray();
+        }
+    }
+
+    private static function processAdditionalFilters($request)
+    {
+        // Process 'term' filter
+        if ($request->has('term')) {
+            self::$requiredFields[] = 'name,%'.$request->input('term').'%,string,like';
+        }
+
+        // Process 'is_all' filter
+        if ($request->input('is_all') === "true") {
+            self::$requiredFields[] =
+                'published_on,'.
+                Carbon::now()
+                    ->subMonth(3)
+                    ->toDateTimeString().
+                ',date,>=';
+        }
+
+        // Process other filters based on 'tabs' input
+        if ($tabs = $request->input('tabs', $request->input('tab', false))) {
```

Comment:
I think this might just have a typo, or was it intentional?
```suggestion
        if ($tabs = $request->input('tabs', false)) {
```

Reply:
It's not typo :)   There are places where we receive tab instead of tabs from the FE

Reply:
Oh perfect, thank you for the clarification